services/wallet/router.go

```
@@ -2,54 +2,276 @@ package wallet
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/status-im/status-go/eth-node/types"
 	"github.com/status-im/status-go/params"
 	"github.com/status-im/status-go/services/wallet/async"
+	"github.com/status-im/status-go/services/wallet/bigint"
+	"github.com/status-im/status-go/services/wallet/bridge"
 	"github.com/status-im/status-go/services/wallet/chain"
+	"github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/transactions"
 )
 
-type SuggestedRoutes struct {
-	Networks []params.Network `json:"networks"`
-}
+type SendType int
 
-func NewRouter(s *Service) *Router {
-	return &Router{s}
+const (
+	Transfer SendType = iota
+	ENSRegister
+	ENSRelease
+	ENSSetPubKey
+	StickersBuy
+)
+const EstimateUsername = "RandomUsername"
+const EstimatePubKey = "0x04bb2024ce5d72e45d4a4f8589ae657ef9745855006996115a23a1af88d536cf02c0524a585fce7bfa79d6a9669af735eda6205d6c7e5b3cdc2b8ff7b2fa1f0b56"
+
+func (s SendType) isTransfer() bool {
+	return s == Transfer
 }
 
-type Router struct {
-	s *Service
+func (s SendType) isAvailableFor(network *params.Network) bool {
+	if s == Transfer {
+		return true
+	}
+
+	if network.ChainID == 1 || network.ChainID == 5 {
+		return true
+	}
+
+	return false
 }
 
-func (r *Router) suitableTokenExists(ctx context.Context, network *params.Network, tokens []*Token, account common.Address, amount float64, tokenSymbol string) (bool, error) {
-	for _, token := range tokens {
-		if token.Symbol != tokenSymbol {
-			continue
+func (s SendType) EstimateGas(service *Service, network *params.Network) uint64 {
+	from := types.Address(common.HexToAddress("0x5ffa75ce51c3a7ebe23bde37b5e3a0143dfbcee0"))
+	tx := transactions.SendTxArgs{
+		From:  from,
+		Value: (*hexutil.Big)(big.NewInt(0)),
+	}
+	if s == ENSRegister {
+		estimate, err := service.ens.API().RegisterEstimate(context.Background(), network.ChainID, tx, EstimateUsername, EstimatePubKey)
+		if err != nil {
+			return 400000
```

I wonder if we should return the error too. EstimateGas generally fails if the transaction will revert, altho we should still give the user the possibility of sending the transaction